src/Lykke.Service.BlockchainApi.Sdk/Controllers/TransactionsController.cs

```diff
@@ -0,0 +1,381 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using Lykke.Common.Log;
+using Lykke.Service.BlockchainApi.Contract;
+using Lykke.Service.BlockchainApi.Contract.Transactions;
+using Lykke.Service.BlockchainApi.Sdk.Domain;
+using Lykke.Service.BlockchainApi.Sdk.Models;
+using Microsoft.AspNetCore.Mvc;
+using MoreLinq;
+using Newtonsoft.Json;
+
+namespace Lykke.Service.BlockchainApi.Sdk.Controllers
+{
+    [ApiController]
+    [Route("/api/transactions")]
+    public class TransactionsController : ControllerBase
+    {
+        readonly IBlockchainApi _api;
+        readonly OperationRepository _operations;
+        readonly AssetRepository _assets;
+        readonly DepositWalletRepository _depositWallets;
+        readonly ILog _log;
+
+        public TransactionsController(IBlockchainApi api, OperationRepository operations,
+            AssetRepository assets, DepositWalletRepository depositWallets, ILogFactory logFactory)
+        {
+            _api = api;
+            _operations = operations;
+            _assets = assets;
+            _depositWallets = depositWallets;
+            _log = logFactory.CreateLog(this);
+        }
+
+        private async Task<ActionResult<BuildTransactionResponse>> Build(Guid operationId, string assetId, 
+            (string from, string fromContext, string to, string amount)[] inOuts, bool includeFee)
+        {
+            var operation = await _operations.GetAsync(operationId);
+
+            if (operation != null && 
+                operation.IsRunning()) 
+            {
+                return Conflict($"Operation is already {operation.GetState()}");
+            }
+
+            var asset = await _assets.GetAsync(assetId);
+
+            if (asset == null)
+            {
+                return BadRequest("Unknown asset");
+            }
+
+            try
+            {
+                // convert string amounts to decimal and validate
+                var actions = inOuts
+                    .Select((a, i) => new OperationAction(i.ToString("D4"), a.from, a.fromContext, a.to, amount: Conversions.CoinsFromContract(a.amount, asset.Accuracy)))
+                    .ToArray();
+
+                if (actions.Any(a => a.Amount <= 0 || !_api.AddressIsValid(a.From) || !_api.AddressIsValid(a.To)))
+                {
+                    return BadRequest("Invalid address(es) and/or negative amount(s)");
+                }
+
+                var separator = _api.GetConstants().PublicAddressExtension?.Separator ?? char.MinValue;
+
+                if (actions.Any(a => a.IsReal(separator)) & 
```

Comment:
Why not use && instead of bit operation?

Reply:
I check if actions of both types are contained in the list, so both conditions must be checked here. Added comment.

Reply:
Implemented XOR-based check.